### ISSUE TITLE
New Feature: S3 control via custodian

### ISSUE THREAD

**Issue by kumarsiyer** [cloud-custodian/cloud-custodian#3146]
Announcement --> https://aws.amazon.com/blogs/aws/amazon-s3-block-public-access-another-layer-of-protection-for-your-accounts-and-buckets/?utm_source=feedburner&utm_medium=feed&utm_campaign=Feed%3A+AmazonWebServicesBlog+%28Amazon+Web+Services+Blog%29

Boto API reference -> https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3control.html

**Comment by kapilt** [comment]
cool, so thinking of structure as an account action as 

```
type: set-s3-public-block
state: boolean
BlockPublicAcls
...
```

**Comment by kumarsiyer** [comment]
We have to do this on account level and individual bucket level.

**Comment by jakauppila** [comment]
This would be a great addition, was trying to look through the schema to find how to do this.

**Comment by gabel0287** [comment]
Seems like this is done for account level public block policies, any updates on being able to set a policy on an individual bucket?

**Comment by atheiman** [comment]
@gabel0287 where do you see how to do this in cloud custodian for an entire account as described here: https://docs.aws.amazon.com/AmazonS3/latest/user-guide/block-public-access-account.html

**Comment by JohnHillegass** [comment]
@kapilt, I believe that this can be closed because of the addition of #5451. I'd update myself but it seems these permissions are locked down a bit, which makes sense.

**Comment by JohnHillegass** [comment]
Closed with addition of #5451
